pkg/utils/securitycontext.go

```diff
@@ -0,0 +1,211 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	v1 "k8s.io/api/core/v1"
+	"keptn-contrib/job-executor-service/pkg/config"
+	"log"
+	"os"
+)
+
+// jobSecurityContextFilePath describes the path of the security config file that is defined in the deployment.yaml
+const jobSecurityContextFilePath = "/config/job-defaultSecurityContext.json"
+
+// podSecurityContextFilePath describes the path of the pod security config file that is defined in the deployment.yaml
+const podSecurityContextFilePath = "/config/job-podSecurityContext.json"
+
+// ReadDefaultJobSecurityContext reads the JSON file defined in jobSecurityContextFilePath and parses the output
+// into a valid v1.SecurityContext data structure
+func ReadDefaultJobSecurityContext() (*v1.SecurityContext, error) {
+	jobSecurityContextBytes, err := os.ReadFile(jobSecurityContextFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var defaultJobSecurityContext v1.SecurityContext
+	err = json.Unmarshal(jobSecurityContextBytes, &defaultJobSecurityContext)
+	if err != nil {
+		return nil, err
+	}
+
+	return &defaultJobSecurityContext, nil
+}
+
+// ReadDefaultPodSecurityContext reads the JSON file defined in podSecurityContextFilePath and parses the output
+// into a valid v1.PodSecurityContext data structure
+func ReadDefaultPodSecurityContext() (*v1.PodSecurityContext, error) {
+	podSecurityContextBytes, err := os.ReadFile(podSecurityContextFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var defaultPodSecurityContext v1.PodSecurityContext
+	err = json.Unmarshal(podSecurityContextBytes, &defaultPodSecurityContext)
+	if err != nil {
+		return nil, err
+	}
+
+	return &defaultPodSecurityContext, nil
+}
+
+// BuildSecurityContext builds the final v1.SecurityContext from the SecurityContext defined in the job configuration
+// and the default job security context. The configs are merged in such a way that the configuration in the task will
+// overwrite each entry that is defined.
+func BuildSecurityContext(defaultSecurityContext *v1.SecurityContext, securityContext config.SecurityContext) *v1.SecurityContext {
```

Comment:
here we could use a trick: serializing to JSON and deserializing into the cloned Security Context which would override the values (need to be tested properly). It would be worth trying to avoid the tedious copying of every field